paddle/fluid/operators/listen_and_serv_op.cc

```diff
@@ -146,15 +143,6 @@ void ListenAndServOp::RunSyncLoop(framework::Executor *executor,
                           recv_scope);
     VLOG(2) << "run all blocks spent " << detail::GetTimestamp() - ts << "(ms)";
 
-    // Reset the received sparse variables, the sum operator would not
-    // sum the input sparse variables which rows is empty at the next
-    // mini-batch.
-    // TODO(Yancey1989): move the reset action into an operator, we couldn't
-    // have any hide logic in the operator.
-    for (framework::Variable *var : sparse_vars) {
```

Comment:
I think this is needed by sparse updates? @Yancey1989

Reply:
Yes, `sparse_vars ` is need by remote sparse update, we need to clear it after each mini-bach, because not each sparse gradient var would been send to the pserver, the clear operation would avoid reuse the old var leaved from  pre-mini-batch.

Reply:
```sparse_vars``` seems always empty because it's not mutated?

Reply:
It would been update if the received var type is `SelectedRows`

https://github.com/PaddlePaddle/Paddle/blob/ff9b1a0f9576bbab8493f1c284768f7135e8e6f8/paddle/fluid/operators/detail/request_handler_impl.cc#L67-L70

Reply:
```sparse_vars_``` is exposed by sparse_vars(), but it seems not called by anyone?

![image](https://user-images.githubusercontent.com/2887803/41077296-6e792420-6a48-11e8-8431-72be455eb289.png)

Reply:
Got it, you're right, seems it's a bug, we need to iterator `sparse_vars_` instead of `sparse_var` which defined in `listen_and_serv_op`.